### PR TITLE
Updates discover logic to check can add token and can query agents

### PR DESCRIPTION
**Pull request by stevenGravy**
Enroll new resources was showing when user didn't have token rights.  It should only show for users who have token create rights and can query the agents.

### CODE REVIEW

File: web/packages/teleport/src/stores/storeUserContext.ts [hunk 1]
```diff
@@ -167,6 +167,6 @@ export default class StoreUserContext extends Store<UserContext> {
   }
 
   hasDiscoverAccess() {
-    return this.hasPrereqAccessToAddAgents() || this.hasAccessToQueryAgent();
+    return this.hasPrereqAccessToAddAgents() && this.hasAccessToQueryAgent();
```
Comment (kimlisa) [inline web/packages/teleport/src/stores/storeUserContext.ts:170]: @ryanclark i vaguely remember someone discussing this, it should've been an AND, but was wondering if there was a reason it was an OR?

Comment (kimlisa) [reply web/packages/teleport/src/stores/storeUserContext.ts:170]: oh wait a minute, this flag is to show the discover feature on our dashboard. i think this is correct (or it's okay to show it), it's a way to "advertise" this feature if you have partial permissions. user should still be able to see it, and we let the user know possible reasons why they can't continue the flow

Comment (stevenGravy) [reply web/packages/teleport/src/stores/storeUserContext.ts:170]: ok, I'll close then

**Comment by stevenGravy** [comment]
Closing as meant to show to advertise feature

File: web/packages/teleport/src/stores/storeUserContext.ts (hunk 1 above)
Comment (scottpgallagher) [reply web/packages/teleport/src/stores/storeUserContext.ts:170]: From an admin perspective I wouldn’t want my users knowing they could add resources themselves even though they actually can’t. I’d be fine with default is we show it but we should have an option to hide it like we do other UI elements. We shouldn’t advertise what you can’t do IMO.

**Pull request closed without merging**